packages/yoastseo/src/worker/AnalysisWebWorker.js

```diff
@@ -1084,6 +1083,11 @@ export default class AnalysisWebWorker {
 
 			const languageProcessor = new LanguageProcessor( this._researcher );
 			this._paper.setTree( build( this._paper.getText(), languageProcessor ) );
+			/*
+			 * We only remove the unwanted elements from the Paper after the HTML tree is built. Removing these elements
+			 * before can lead to incorrectly calculating the positions of sentences and tokens in the tree.
+			 */
+			paper._text = removeHtmlBlocks( paper._text );
```

Comment:
Now when I see this again, isn't this supposed to be `this._paper._text = removeHtmlBlocks( paper._text );` ? 

This is because at this point we've already assigned `paper` to `this._paper above it.